src/main/java/org/pentaho/mongo/wrapper/MongoWrapperUtil.java

```diff
@@ -96,6 +97,11 @@ public static MongoClientWrapper createMongoClientWrapper( MongoProperties.Build
 
   private static void setIfNotNullOrEmpty( MongoProperties.Builder builder, MongoProp prop, String value ) {
     if ( value != null && value.trim().length() > 0 ) {
+      boolean isPassword = MongoProp.PASSWORD.equals( prop );
+      boolean isEncrypted = value.startsWith( Encr.PASSWORD_ENCRYPTED_PREFIX );
```

Comment:
Shouldn't do this line - it's handled by the ...Optionally... bit in the Encr class. https://github.com/pentaho/pentaho-kettle/blob/master/core/src/org/pentaho/di/core/encryption/KettleTwoWayPasswordEncoder.java#L86-L91